python/pyarrow/fs.py

```diff
@@ -63,6 +63,32 @@ def __getattr__(name):
     )
 
 
+def _ensure_filesystem(filesystem):
+    if isinstance(filesystem, FileSystem):
+        return filesystem
+
+    # handle fsspec-compatible filesystems
+    try:
+        import fsspec
+    except ImportError:
+        pass
+    else:
+        if isinstance(filesystem, fsspec.AbstractFileSystem):
+            if type(filesystem).__name__ == 'LocalFileSystem':
+                # In case its a simple LocalFileSystem, use native arrow one
+                return LocalFileSystem()
+            return PyFileSystem(FSSpecHandler(filesystem))
+
+    # map old filesystems to new ones
+    from pyarrow.filesystem import LocalFileSystem as LegacyLocalFileSystem
+
+    if isinstance(filesystem, LegacyLocalFileSystem):
+        return LocalFileSystem()
+    # TODO handle HDFS?
```

Comment:
Do you have any idea about this one? The "legacy" one defined in `io-hdfs.pxi` is backed by the C++ `arrow::io::HadoopFileSystem`, and the new filesystem is backed by `arrow::fs::HadoopFileSystem` which itself holds a `arrow::io::HadoopFileSystem` as "client". 
So naively I would say (with some additional methods), one should be able to make a new HadoopFileSystem by wrapping the C++ "client" of the legacy filesystem?

Reply:
Perhaps, but it's also not very important. Let's not bother about it until there's a real need for it.